scripts/apitypings/main.go

```diff
@@ -57,6 +59,14 @@ func main() {
 	}
 
 	_, _ = fmt.Print("// Code generated by 'make site/src/api/typesGenerated.ts'. DO NOT EDIT.\n\n")
+
+	countriesTS, err := generateSliceOfObjectsTypeScript("countries", codersdk.Countries)
+	if err != nil {
+		log.Fatal(ctx, fmt.Sprintf("generate countries typeScript: %s", err.Error()))
+	}
+	_, _ = fmt.Print("// The code below is generated from codersdk/countries.go.\n")
+	_, _ = fmt.Print(countriesTS, "\n")
+
```

Comment:
To do this correctly, we should not have to specify `codersdk.Countries`. We should implement it here:

https://github.com/coder/coder/blob/afacb07140a68f10f9507591340484b80235a038/scripts/apitypings/main.go#L458-L459


I feel a bit apprehensive to import `codersdk` into this tool. 

We could make it a new generator (:cry:) or actually auto-gen vars from the codersdk package (like `codersdk.Me`).

If we do the `vars`, we might want to have some opt-in comment. Idk how many var declarations we have

Reply:
We should really have a general autogen tool and pattern for these. When the types are in Go and no package AST parsing is necessary.

`rbacgen` is an example.

Reply:
@Emyrk 

got it, will implement a general autogen tool. As suggested, I'm thinking of doing it similar to how `rbacgen` is done.

So the autogen tool will be under `scripts/valuegen` which contains `main.go` which is the script itself and `typescript.tstmpl` which will be the TS template to which we will convert the `countries.go`

https://github.com/coder/coder/blob/1702c64312fb9b03e534c935dc62971a3243856c/codersdk/countries.go#L3

If similar types conversion needs are needed in the future for other types in Go, it can be handled in the script to use the same `typescriptconst.tstmpl` template or a different one based on the need.

does that make sense? wdyt?

Reply:
@joobisb That all sounds reasonable. It'll probably throw into a different `.ts` file.

It would be awesome if `rbacgen` and `valuegen` could be merged into 1 :smile:

Reply:
@Emyrk I have updated the scripts and have merged both into a single `typegen` script, please do have a look